templates/setup.py.jj2

```diff
@@ -112,7 +112,15 @@ if python_implementation == "PyPy":
 INSTALL_REQUIRES = [
 {% for dependency in dependencies: %}
   {% if ';' not in dependency and not dependency.startswith('#'): %}
+    {% if '#' in dependency: %}
+      {% if '+' in dependency: %}
+      {% set dependency = dependency.split('#') %}
```

Comment:
Rewrite.  The only known case that `#` is part of the requirement is when it is `#egg` .  Write your logic around that, making it much more precise and readable .

Any other use of `#` is then a comment.

p.s. this line should have been indented